From 882347c78447e5a229f8a9952df4f36771f65cde Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Mon, 27 Mar 2023 08:41:33 +0200
Subject: [PATCH 0196/1695] dts: st: enable RCC driver in stm32mp251

Add the RCC node to support clock on stm32mp25 platform.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git c828f4cfef11054f3190b0311dcaea59f5ee1570 dts: st: enable RCC driver in stm32mp251]
Change-Id: I5213efcf44786a8e5a481b93b32e133065118666
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
---
 core/arch/arm/dts/stm32mp251.dtsi | 36 ++++++++++++++++++++++++++++++-
 1 file changed, 35 insertions(+), 1 deletion(-)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 0a11f50d1..858aad7e5 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -5,6 +5,8 @@
  */
 
 #include <dt-bindings/interrupt-controller/arm-gic.h>
+#include <dt-bindings/clock/stm32mp25-clks.h>
+#include <dt-bindings/reset/stm32mp25-resets.h>
 
 / {
 	#address-cells = <2>;
@@ -257,8 +259,28 @@
 		};
 
 		rcc: rcc@44200000 {
+			compatible = "st,stm32mp25-rcc", "syscon";
 			reg = <0x44200000 0x10000>;
-			status = "disabled";
+			interrupts = <GIC_SPI 259 IRQ_TYPE_LEVEL_HIGH>;
+
+			#clock-cells = <1>;
+			#reset-cells = <1>;
+			clocks = <&clk_hse>, <&clk_hsi>, <&clk_lse>,
+				 <&clk_lsi>, <&clk_msi>, <&clk_i2sin>;
+			clock-names = "clk-hse", "clk-hsi", "clk-lse",
+				      "clk-lsi", "clk-msi", "clk-i2sin";
+
+			hsi_calibration: hsi-calibration {
+				compatible = "st,hsi-cal";
+				st,cal_hsi_dev = <31>;
+				st,cal_hsi_ref = <1953>;
+				status = "disabled";
+			};
+
+			msi_calibration: msi-calibration {
+				compatible = "st,msi-cal";
+				status = "disabled";
+			};
 		};
 
 		pwr: pwr@44210000 {
@@ -289,6 +311,7 @@
 				interrupt-controller;
 				#interrupt-cells = <2>;
 				reg = <0x0 0x400>;
+				clocks = <&rcc CK_BUS_GPIOA>;
 				st,bank-name = "GPIOA";
 				status = "disabled";
 			};
@@ -299,6 +322,7 @@
 				interrupt-controller;
 				#interrupt-cells = <2>;
 				reg = <0x10000 0x400>;
+				clocks = <&rcc CK_BUS_GPIOB>;
 				st,bank-name = "GPIOB";
 				status = "disabled";
 			};
@@ -309,6 +333,7 @@
 				interrupt-controller;
 				#interrupt-cells = <2>;
 				reg = <0x20000 0x400>;
+				clocks = <&rcc CK_BUS_GPIOC>;
 				st,bank-name = "GPIOC";
 				status = "disabled";
 			};
@@ -319,6 +344,7 @@
 				interrupt-controller;
 				#interrupt-cells = <2>;
 				reg = <0x30000 0x400>;
+				clocks = <&rcc CK_BUS_GPIOD>;
 				st,bank-name = "GPIOD";
 				status = "disabled";
 			};
@@ -329,6 +355,7 @@
 				interrupt-controller;
 				#interrupt-cells = <2>;
 				reg = <0x40000 0x400>;
+				clocks = <&rcc CK_BUS_GPIOE>;
 				st,bank-name = "GPIOE";
 				status = "disabled";
 			};
@@ -339,6 +366,7 @@
 				interrupt-controller;
 				#interrupt-cells = <2>;
 				reg = <0x50000 0x400>;
+				clocks = <&rcc CK_BUS_GPIOF>;
 				st,bank-name = "GPIOF";
 				status = "disabled";
 			};
@@ -349,6 +377,7 @@
 				interrupt-controller;
 				#interrupt-cells = <2>;
 				reg = <0x60000 0x400>;
+				clocks = <&rcc CK_BUS_GPIOG>;
 				st,bank-name = "GPIOG";
 				status = "disabled";
 			};
@@ -359,6 +388,7 @@
 				interrupt-controller;
 				#interrupt-cells = <2>;
 				reg = <0x70000 0x400>;
+				clocks = <&rcc CK_BUS_GPIOH>;
 				st,bank-name = "GPIOH";
 				status = "disabled";
 			};
@@ -369,6 +399,7 @@
 				interrupt-controller;
 				#interrupt-cells = <2>;
 				reg = <0x80000 0x400>;
+				clocks = <&rcc CK_BUS_GPIOI>;
 				st,bank-name = "GPIOI";
 				status = "disabled";
 			};
@@ -379,6 +410,7 @@
 				interrupt-controller;
 				#interrupt-cells = <2>;
 				reg = <0x90000 0x400>;
+				clocks = <&rcc CK_BUS_GPIOJ>;
 				st,bank-name = "GPIOJ";
 				status = "disabled";
 			};
@@ -389,6 +421,7 @@
 				interrupt-controller;
 				#interrupt-cells = <2>;
 				reg = <0xa0000 0x400>;
+				clocks = <&rcc CK_BUS_GPIOK>;
 				st,bank-name = "GPIOK";
 				status = "disabled";
 			};
@@ -417,6 +450,7 @@
 				interrupt-controller;
 				#interrupt-cells = <2>;
 				reg = <0 0x400>;
+				clocks = <&rcc CK_BUS_GPIOZ>;
 				st,bank-name = "GPIOZ";
 				st,bank-ioport = <11>;
 				status = "disabled";
-- 
2.47.2

